src/stream/src/executor/over_window/over_partition.rs

```diff
@@ -530,12 +536,19 @@ impl<'a, S: StateStore> OverPartition<'a, S> {
                 let (key, row) = curr_key_cursor
                     .key_value()
                     .expect("cursor must be valid until `last_curr_key`");
+                let mut should_continue = true;
+
                 let output = states.slide_no_evict_hint()?;
                 compute_count += 1;
 
                 let old_output = &row.as_inner()[input_schema_len..];
                 if !old_output.is_empty() && old_output == output {
                     same_output_count += 1;
+
+                    if rank_funcs_only && key >= last_delta_key {
```

Comment:
Will the rank always be sorted in ascending order?

Reply:
The key is memcmp-encoded, which means the order type is considered.